### PR TITLE
Code merging

### PR DESCRIPTION
**Pull request by stelladk**
- Documentation updates
- More unit tests
- Typing fixes
- feature: disable logging

### CODE REVIEW

**Comment by stelladk** [comment]
Is the .pre-commit-config.yaml file used by CI different than the one in the project?

**Review by sylvchev — COMMENTED** [review]
nice additions.

File: pyproject.toml [hunk 1]
```diff
@@ -25,6 +25,7 @@ dependencies = [
     "numpy",
     "matplotlib",
     "pyvis",
+    "fvcore",
```
[the remark this answers is not in the document]
Comment (sylvchev) [reply pyproject.toml:28]: we need to add tochmetrics as well to pass the tests.

File: .gitignore [hunk 2]
```diff
@@ -169,7 +169,16 @@ cython_debug/
 #  be found at https://github.com/github/gitignore/blob/main/Global/JetBrains.gitignore
 #  and can be added to the global gitignore or merged into this file.  For a more nuclear
 #  option (not recommended) you can uncomment the following to ignore the entire idea folder.
-.idea
```
[the remark this answers is not in the document]
Comment (stelladk) [reply .gitignore:172]: that's by accident

File: pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (stelladk) [reply pyproject.toml:28]: I am working to remove the torchmetrics dependency

**Comment by stelladk** [comment]
Can we have a set of dependencies for debug?

File: src/gromo/script/run.sh
Comment (sylvchev) [inline src/gromo/script/run.sh:1]: I think you could keep this kind of files, related with local clusters, on your cluster account or on a separate repo. They are too specific for an infrastructure to get in the main the repo IMO

File: src/gromo/script/test_coverage.sh
Comment (sylvchev) [inline src/gromo/script/test_coverage.sh:1]: no need for this script, you could see the coverage on https://app.codecov.io/github/growingnet/gromo

File: src/gromo/utils/utils.py [hunk 3]
```diff
@@ -283,3 +287,106 @@ def size_to_color(size):
...
+        true positives, false positives, false negatives
+    """
+    true_positives = np.sum((actual == label) & (predicted == label))
+    false_positives = np.sum((actual != label) & (predicted == label))
+    false_negatives = np.sum((predicted != label) & (actual == label))
+
+    return true_positives, false_positives, false_negatives
+
+
+def f1(actual: torch.Tensor, predicted: torch.Tensor, label: int) -> float:
```
Comment (sylvchev) [inline src/gromo/utils/utils.py:318]: f1 is already implemented in torcheval, it is still marked as unstable but very actively supported by Meta. There is also sklearn that proposed a very robust implementation (need to transform Tensor into ndarray).
Do you have a specific reason to re-implement existing code?

**Review by sylvchev — COMMENTED** [review]
We could push tests later as you explained me.

File: src/gromo/utils/utils.py (hunk 3 above)
Comment (stelladk) [reply src/gromo/utils/utils.py:318]: I was using torchmetrics initially and removed it to reduce dependencies

File: docs/source/whats_new.rst [hunk 4]
```diff
@@ -18,6 +18,9 @@ Develop branch
 Enhancements
 ~~~~~~~~~~~~
 
+- Unified documentation style (:gh:`14` by `Stella Douka`_)
+- Updated Unit Tests (:gh:`14` by `Stella Douka`_)
+- Option to disable logging (:gh: `14` by `Stella Douka`_)
```
Comment (sylvchev) [inline docs/source/whats_new.rst:23]:
```suggestion
- Option to disable logging (:gh:`14` by `Stella Douka`_)
```

File: src/gromo/utils/utils.py (hunk 3 above)
Comment (sylvchev) [reply src/gromo/utils/utils.py:344]: ok, we will have to discuss if we rely on external libs for evaluation metrics. The balance is always hard to find between avoiding dependencies and recoding everything, that is adding bugs.

**Review by sylvchev — APPROVED** [review]

**Comment by sylvchev** [comment]
@stelladk Thanks 🙏

**Pull request merged**